src/main/scala/com/mwz/sonar/scala/Metadata.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2018-2020  All sonar-scala contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.mwz.sonar.scala
+
+import java.nio.file.Paths
+
+import cats.data.Chain
+import cats.effect.Blocker
+import cats.effect.ExitCode
+import cats.effect.IO
+import cats.effect.IOApp
+import cats.syntax.functor._
+import com.mwz.sonar.scala.metadata._
+import com.mwz.sonar.scala.metadata.scalastyle.ScalastyleRules
+import com.mwz.sonar.scala.metadata.scalastyle.ScalastyleRulesRepository
+import fs2.Stream
+import fs2.io.file._
+import fs2.text
+import io.circe.Printer
+import io.circe.generic.auto._
```

Comment:
Wouldn't be better to use `semiauto`?

Reply:
Well, I'd have to add `@JsonCodec` annotations to all of those case classes or add implicit encoders, so using auto is easier in this case, especially that I don't need any custom name mapping etc. Why do you think semiauto would be better?

Reply:
Catching the instances so they do not get recreated all over again. In general `auto` is not recommended as it may consume too much compilation time.
But yeah, it is easier and given the project is small, it probably isn't that bad.

Reply:
ok, I don't want to stand in a way of faster compile times - changed it to semiauto 😃
"every little helps"